db/revision_cache_interface.go

```diff
@@ -259,6 +267,22 @@ func revCacheLoader(ctx context.Context, backingStore RevisionCacheBackingStore,
 	return revCacheLoaderForDocument(ctx, backingStore, doc, id.RevID)
 }
 
+func revCacheLoaderForCv(ctx context.Context, backingStore RevisionCacheBackingStore, id IDandCV, unmarshalBody bool) (bodyBytes []byte, body Body, history Revisions, channels base.Set, removed bool, attachments AttachmentsMeta, deleted bool, expiry *time.Time, err error) {
```

Comment:
Where are we checking that the current version of the document that we get from backingStore matches cv?

Reply:
A unit test for this case would be a good idea, as this seems like an easy place for subtle bugs to get introduced.

Reply:
Good point I've added this check and appropriate error message to be returned in that fail case. Also added a unit test to cover this scenario entitled `TestLoaderMismatchInCV`